jax/experimental/sparse/ad.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, Sequence, Tuple, Union
+
+import numpy as np
+
+import jax
+from jax import core
+from jax import tree_util
+from jax.api_util import _ensure_index, _ensure_index_tuple
+from jax.util import safe_zip
+from jax._src.util import wraps
+from jax._src.traceback_util import api_boundary
+from .ops import BCOO
+
+
+def value_and_grad(fun: Callable, argnums: Union[int, Sequence[int]] = 0, **kwargs) -> Callable[..., Tuple[Any, Any]]:
```

Comment:
Could you try to wrap to 80 characters? It's not a strict rule but it helps some of our colleagues. This line seems super long. (I have a command in my editor which hard-wraps to 80 chars. Maybe yours can do that too!)

Reply:
Done